File: backend/Models/Product.cs
namespace ProductAPI.Models;

public class Product
{
    public string Name { get; set; } = string.Empty;
    public double PopularityScore { get; set; }
    public double Weight { get; set; }
    public Dictionary<string, string> Images { get; set; } = new();
    public decimal Price { get; set; }
    public double Rating { get; set; }
}

public class ProductData
{
    public string Name { get; set; } = string.Empty;
    public double PopularityScore { get; set; }
    public double Weight { get; set; }
    public Dictionary<string, string> Images { get; set; } = new();
}

File: frontend/src/types/Product.ts
export interface Product {
  name: string;
  popularityScore: number;
  weight: number;
  images: {
    yellow: string;
    rose: string;
    white: string;
  };
  price: number;
  rating: number;
}

export type ColorOption = 'yellow' | 'rose' | 'white';
